ibis/backends/pyspark/tests/test_basic.py

```diff
@@ -214,3 +215,25 @@ def test_alias_after_select(client):
 
     result = table.compile().toPandas()
     tm.assert_series_equal(result['id'], result['id2'], check_names=False)
+
+
+@pytest.mark.parametrize(
+    ('selection_fn', 'expected'),
+    [
+        (lambda t: t[['id']], [True]),
+        (lambda t: t.mutate(v=t['id']), [False, False]),
```

Comment:
What does the two boolean value refer to? I only saw one column expr inside mutate

Reply:
Yeah it's a little confusing because depending on the operation being done, there might be more than one selection in the Selection op. For example:

`t.mutate(v=t['id'])` <-- first selection is the TableExpr, second is the ColumnExpr(v)
`t.mutate(id=t['str_col'])` <-- since this is an overwrite, ibis expands this out into its constituent table columns, so the first is id and the second is str_col.

I can add a comment to explain what the variable-length bool array means in general.

Reply:
I see. I was hoping to make the test a bit clearer than that. e.g.

```
# new column id is selections[1]
lambda t: t.mutate(id=t['id']).op().selections[1], True
# new column id is selections[1], cannot be replaced since new id != t['id']
lambda t: t.mutate(id=t['id']) + 1.op().selections[1], False
```

Reply:
I think inside the test function we need to have access to the root table itself after the selection/mutation so we can pass that into `_can_be_replaced_by_column_name`, so I added the index of the selection to test as an additional parameter.